ui/pages/confirmations/components/confirm/row/dataTree.tsx

```diff
@@ -72,18 +78,53 @@ const DataField = memo(
         <DataTree
           data={value}
           isPermit={isPermit}
+          primaryType={primaryType}
           tokenDecimals={tokenDecimals}
         />
       );
     }
-    if (isPermit && label === 'value') {
+
+    const isPermitBatchOrSingle =
+      primaryType === PERMIT_PRIMARY_TYPE.PERMIT_BATCH ||
+      primaryType === PERMIT_PRIMARY_TYPE.PERMIT_SINGLE;
+
+    const isPermitTransferFrom =
+      primaryType === PERMIT_PRIMARY_TYPE.PERMIT_BATCH_TRANSFER_FROM ||
+      primaryType === PERMIT_PRIMARY_TYPE.PERMIT_TRANSFER_FROM;
+
+    const isOrder =
+      primaryType === PERMIT_PRIMARY_TYPE.ORDER ||
+      primaryType === PERMIT_PRIMARY_TYPE.ORDER_COMPONENTS;
+
+    const isDate =
+      value &&
+      ((label === 'deadline' && isPermit) ||
+        (label === 'endTime' && isOrder) ||
+        (label === 'expiration' && isPermitBatchOrSingle) ||
+        (label === 'sigDeadline' &&
+          (isPermitBatchOrSingle || isPermitTransferFrom)) ||
+        (label === 'startTime' && isOrder) ||
+        (label === 'validTo' && isOrder));
```

Comment:
As this grows, could we instead migrate it to a constant such as:

```
const DATE_FIELDS_BY_PRIMARY_TYPE = {
    [PrimaryType.PermitSingle]: ['expiration']
    ...
}
```

I appreciate there's a little duplication between types but maybe easier to maintain?

Reply:
can do 👍🏼  [5f8e122](https://github.com/MetaMask/metamask-extension/pull/26107/commits/5f8e122de845dce87852ee7d72a1814fa285bc57)